src/lib/comms/include/sol-netctl.h

```diff
@@ -553,6 +614,67 @@ sol_netctl_find_service_by_name(const char *service_name)
 }
 
 /**
+ * @brief register a agent for network connection
+ *
+ * A single agent is registered for an application that registering a new agent.
+ * unregister the previous and to unregister the last one you should use NULL.
+ *
+ * @see struct sol_netctl_agent
+ *
+ * @param agent the agent Callback struct to be called when the related information is updated.
+ * NULL is used to unregister the agent.
+ * @param data Add a user data per callback.
+ *
+ * @return 0 on success, -errno on failure.
+ */
+int
+sol_netctl_register_agent(const struct sol_netctl_agent *agent, const void *data);
```

Comment:
I'm wondering whether this function is too complicated or not. I mean, this function has 3 different behaviours

1- Register a new agent when there is no previous agent registered - Success
2- Register a new agent when there is an agent - Fail
3- Unregister an agent passing NULL - What will return if there is no agent registered ?

Maybe it's worth add a separated function to unregister the agent.